src/WebJobs.Script/Metrics/HostMetricsCollector.cs

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+#nullable enable
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Diagnostics.Metrics;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.WebJobs.Script.Metrics
+{
+    public class HostMetricsCollector : IHostMetricsCollector, IDisposable
+    {
+        private readonly MeterListener _meterListener;
+        private readonly ILogger<HostMetricsCollector> _logger;
+        private ConcurrentDictionary<string, long> _metricsCache = new();
+
+        public HostMetricsCollector(ILogger<HostMetricsCollector> logger)
+        {
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+            _meterListener = new();
+
+            Start();
+        }
+
+        public void Start()
+        {
+            _meterListener.InstrumentPublished = (instrument, listener) =>
+            {
+                if (instrument.Meter.Name is MetricConstants.Host)
+                {
+                    listener.EnableMeasurementEvents(instrument);
+                }
+            };
+
+            _meterListener.SetMeasurementEventCallback<long>(OnMeasurementRecordedLong);
+            _meterListener.Start();
+        }
+
+        internal void OnMeasurementRecordedLong(
+            Instrument instrument,
+            long measurement,
+            ReadOnlySpan<KeyValuePair<string, object?>> tags,
+            object? state)
+        {
+            if (instrument == null)
+            {
+                throw new ArgumentNullException(nameof(instrument));
+            }
+
+            _logger.LogDebug("{name} recorded measurement {measurement}", instrument.Name, measurement);
```

Comment:
Do we need this log? Seems like this will produce unnecessary log noise.

Reply:
Nope, was just using it for testing. Will remove